api/client.go

```diff
@@ -657,19 +676,47 @@ func (c *Client) Token() string {
 
 // SetToken sets the token directly. This won't perform any auth
 // verification, it simply sets the token properly for future requests.
+// Setting a token will kill any routine polling a sink for a token.
 func (c *Client) SetToken(v string) {
-	c.modifyLock.Lock()
-	defer c.modifyLock.Unlock()
-
-	c.token = v
+	switch {
+	// no op
+	case v == "" && c.config.TokenFileSinkPath == "":
```

Comment:
We still want to update c.token in this branch.

Reply:
to be `""`?